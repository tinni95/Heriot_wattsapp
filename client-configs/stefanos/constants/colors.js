export const main = "#301D5C";
export const secondary = "#E0A749";

export const itemCard = {
    background: "#F4F4F4",
    plusBtnBckgrnd: "#348653",
    plusBtnText: "#FFFFFF",
};

export const searchBar = {
    background: secondary,
};
