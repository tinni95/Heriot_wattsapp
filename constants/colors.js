export const main = "#301D5C";
export const secondary = "#E0A749";
export const white = "#fff";
export const transparent = "rgba(255, 255, 255, 0.0)";

export const searchBar = {
    background: secondary,
};

export const navigationBar = {
    underlayColor: "#f0f4f7",
    navBarTextItem: "#D6D1D1",
    selectedNavBorderBottom: "#E0A749",
};
