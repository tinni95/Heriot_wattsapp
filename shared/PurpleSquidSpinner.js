import { ActivityIndicator, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import React from "react";
import { main } from "../constants/colors";

const PurpleSquidSpinner = ({ color }) => (
    <ActivityIndicator
        color={color}
        size="large"
        style={styles.activityIndicator}
    />
);

PurpleSquidSpinner.propTypes = {
    color: PropTypes.string,
};

PurpleSquidSpinner.defaultProps = {
    color: main,
};

const styles = StyleSheet.create({
    activityIndicator: {
        flex: 1,
        justifyContent: "flex-start",
    },
});

export default PurpleSquidSpinner;
