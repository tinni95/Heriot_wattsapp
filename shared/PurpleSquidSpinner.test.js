import React from "react";
import { render } from "react-native-testing-library";

import { main } from "../constants/colors";

import PurpleSquidSpinner from "./PurpleSquidSpinner";

describe("PurpleSquidSpinner", () => {
    describe("with no non-default props", () => {
        it("renders successfully to match snapshot", () => {
            const rendered = render(<PurpleSquidSpinner />);
            expect(rendered.toJSON()).toBeTruthy();
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });

    describe("with a different color prop", () => {
        it("renders successfully to match snapshot", () => {
            const rendered = render(<PurpleSquidSpinner color={main} />);
            expect(rendered.toJSON()).toBeTruthy();
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
