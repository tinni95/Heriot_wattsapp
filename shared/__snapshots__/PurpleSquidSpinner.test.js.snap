// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurpleSquidSpinner with a different color prop renders successfully to match snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#301D5C"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
/>
`;

exports[`PurpleSquidSpinner with no non-default props renders successfully to match snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#301D5C"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
/>
`;
